storagenode/console/consoleserver/server.go

```diff
@@ -65,10 +65,19 @@ func NewServer(logger *zap.Logger, config Config, service *console.Service, list
 
 	// handle static pages
 	if config.StaticDir != "" {
+		// a specific directory has been configured. use it
 		fs = http.FileServer(http.Dir(server.config.StaticDir))
+	} else if consoleassets.FileSystem != nil {
+		// assets are compiled into this binary, use them
+		fs = http.FileServer(consoleassets.FileSystem)
+	}
 
+	if fs != nil {
 		mux.Handle("/static/", http.StripPrefix("/static", fs))
-		mux.Handle("/", http.HandlerFunc(server.appHandler))
+		mux.Handle("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			r.URL.Path = "/dist/"
```

Comment:
Handlers should not modify the request, https://golang.org/pkg/net/http/#Handler

Reply:
whoops, you're right, i thought http.StripPrefix modified but I see it makes a copy. will need to make a copy of the http.Request here first. thx

Reply:
I don't get this change does it the same as the before `apphandler`?

This question doesn't say anything about what @egonelbre commented.

Reply:
this should be

```
mux.Handle("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
       r = r.Clone(r.Context())
       r.URL.Path = "/dist/"
       fs.ServeHTTP(w, r)
}))
```
the change is that we use `fs` directly, which might not be using `config.StaticDir` anymore

Reply:
yes, that's the change that I've done, however, I was trying to understand if this change (using the `fs`) is doing the same that the old `appHandler`.

Reply:
no, the old appHandler always hit the filesystem. this new way will use go-bindata if appropriate

Reply:
OK, I wrongly understood that `http.ServeFile(w, r, filepath.Join(server.config.StaticDir, "dist", "index.html"))` was serveing just `dist/index.html` file.